src/js/legend.js

```diff
@@ -0,0 +1,198 @@
+
+    gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors = function(dataSet, colors) {
+        var c = [];
+        for(var i=0; i<dataSet.length; i++) {
+            var d = {
+                id: dataSet[i].id,
+                label: dataSet[i].label,
+                value: dataSet[i].value,
+                color: colors(i)
+            };
+            c.push(d);
+        }
+        return c;
+    };
+
+    gpii.chartAuthoring.pieChart.legend.draw = function(that) {
+
+        var table = that.table,
+            l = that.options.legendOptions,
+            dataSet = gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors(that.model.dataSet,that.color),
+            rowClass = that.classes.row,
+            legendColorCellClass = that.classes.legendColorCell,
+            legendLabelCellClass = that.classes.labelCell,
+            legendValueCellClass = that.classes.valueCell,
+            sort = l.sort;
+        var tbody = table.selectAll("tbody");
+
+        var rows = tbody.selectAll("tr")
+            .data(dataSet, function(d) {
+                return d.id;
+            });
+
+        var addedRows = rows.enter().append("tr");
+
+
+        addedRows
+        .attr({
+            "class": rowClass
+        });
+
+        addedRows
+        .append("td")
+        .attr({
+            "style": function(d, i) {
+                return "background-color: "+d.color+";";
+            },
+            "class": legendColorCellClass
+        });
+
+        addedRows
+        .append("td")
+        .attr({
+            "class": legendLabelCellClass
+        })
+        .html(function(d, i) {
+            return d.label;
+        });
+
+        addedRows.append("td")
+        .attr({
+            "class": legendValueCellClass
+        })
+        .html(function(d,i) {
+            return d.value;
+        });
+
+
+        var removedRows = rows.exit();
+
+        removedRows.remove();
+
+
+        if(sort === true) {
+            rows.sort(function(a,b){
+                return b.value - a.value;
+            });
+        }
+
+    };
+
+    gpii.chartAuthoring.pieChart.legend.create = function(that) {
+        var container = that.container,
+            dataSet = that.model.dataSet,
+            l = that.options.legendOptions,
+            colors = l.colors,
+            tableClass = that.classes.table;
+
+        if (dataSet.length === 0) {
+            return;
+        }
+
+        that.color = colors ? d3.scale.ordinal().range(colors) : d3.scale.category10();
+
+        that.table = that.jQueryToD3(container)
+            .append("table")
+            .attr({
+            "class": tableClass
+        });
+
+        that.table.append("thead");
+        that.table.append("tbody");
+
+        var thead = that.table.selectAll("thead");
+
+        thead.append("th")
+            .attr({
+                "scope":"col"
+            })
+            .html("Legend");
```

Comment:
The static messages to be shown on UI need to go into "strings" option block so they are user configurable.

The doc: http://docs.fluidproject.org/infusion/development/ComponentConfigurationOptions.html#strings
The example: https://github.com/fluid-project/infusion/blob/master/src/components/overviewPanel/js/OverviewPanel.js#L141

Same comment applies to line 184 and 190.